### ISSUE TITLE
Question of Kernel Extension Behavior

### ISSUE THREAD

**Issue by apizz** [erikberglund/ProfileCreator#157]
Posting from [Slack](https://macadmins.slack.com/archives/CH9LT65U6/p1556825685065000):

IIRC, if you have a profile and you want to whitelist specific kernel extensions you can’t do this for some you have to do it for all managed kexts.  Am I correct in thinking that you can’t approve all kexts for some Team IDs and approve only individual kexts for others in the same profile?

If this is the case, some updates should be made to the Kernel Extension manifest, both in description and `pfm_exclude` behavior

**Comment by erikberglund** [comment]
I don't have any information about this, but I don't completely understand it either. Are you saying if you whitelist a team id, then only want to whitelist some specific extension from that team id but not all? Or that only one payload can whitelist one teamid? Do you have an example? Or where this has been an issue?

**Comment by apizz** [comment]
This could my misremembering, or incorrectly hearing from others, but if you create a KEXT profile and you allow a team ID and team ID with specific kernel extensions in the same profile, you get something like this:
```
<key>AllowedKernelExtensions</key>
<dict>
	<key>ABCD123456</key>
	<array>
		<string>com.apple.kexttest</string>
	</array>
</dict>
<key>AllowedTeamIdentifiers</key>
<array>
	<string>EFGH123456</string>
</array>
```
My understanding was that because you have a TeamID which is more restrictive because you've specified KEXT bundle IDs from the TeamID that are permitted (`AllowedKernelExtensions`), that this means anything which you only specify a TeamID for (`AllowedTeamIdentifiers`) doesn't get applied.

Using Jamf as an example, you're able to specify both a TeamID and a TeamID with individual approved kexts in a profile, but I don't know if this still produces the desired result.  If it did not, it would require a `pfm_exclude` such that `AllowedKernelExtensions` & `AllowedTeamIdentifiers` couldn't be used in the same profile.

**Comment by apizz** [comment]
https://macadmins.slack.com/archives/CC84NA6KE/p1557174436073500

**Comment by erikberglund** [comment]
Ok, I see. So the ideal would be to not allow any team id defined in `AllowedTeamIdentifiers` to be used in `AllowedKernelExtensions`. That is a very specific change to just this payload, I'm not sure how to handle that right now. We should start by adding a note to the AllowedKernelExtensions describing this issue.

**Comment by apizz** [comment]
From my point of view, it's less about preventing the same TeamID from being present in both preferences and preventing both preferences from being added in the same profile.  Thus my previous thought about adding a `pfm_exclude`

**Comment by apizz** [comment]
Hmmm, I hadn't thought of that.  I think a `pfm_note` would be sufficient to handle that part of it.

From different conversations I've had in the last year, I don't know if there (m)any environments that deploy multiple KEXT profiles.  We deploy just one because similar to PPPCP, if we're whitelisting something we're doing it for all our machines, and not managing separate profiles for different subgroups.

**Comment by apizz** [comment]
Resolved in v0.3.0 release with additional wording
